lib/ab_test.rb

```diff
@@ -0,0 +1,96 @@
+class AbTest
+  extend Forwardable
+
+  attr_reader :allowed_variants, :expires
+  def_delegators :@ab_test, :ab_test_name, :dimension, :control_variant
```

Comment:
This is interesting.
Hadn't come across [def_delegators](https://ruby-doc.org/stdlib-2.7.2/libdoc/forwardable/rdoc/Forwardable.html#method-i-def_delegators) before.

So... here we take an instance of `Abtest`, which might be assinged to a variable eg `test`.

This means that if you call `test.ab_test_name`, it'll alias it to call that from the `@ab_test` object that is created and stored in the internal state of `test`?

Did i get that right?
That's cool.